### PR TITLE
[Chuong] function search offline

### PR DESCRIPTION
**Pull request by onstonboy**
done

### CODE REVIEW

File: app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchOfflineFragment.java [hunk 1]
```diff
@@ -25,13 +44,68 @@ public void onCreate(Bundle savedInstanceState) {
...
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                if (!mEditText.getText().toString().isEmpty()) {
+                    mPresenter.doPassKeyWord();
+                } else {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchOfflineFragment.java]: ERROR: Empty else block.

File: app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchOfflinePresenter.java [hunk 2]
```diff
@@ -1,5 +1,14 @@
 package com.fstyle.androidtrainning.screen.searchoffline;
 
+import android.content.Context;
+import android.util.Log;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchOfflinePresenter.java]: ERROR: Unused import: android.util.Log. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchOfflinePresenter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.util.Log'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchOfflineFragment.java [hunk 3]
```diff
@@ -45,4 +119,123 @@ public void onStop() {
...
+    public void onGetListAlbumSuccess(List<Album> album) {
+        if (album == null) {
+            return;
+        }
+        mTxtLabelAlbum.setVisibility(View.VISIBLE);
+        List<Album> list = new ArrayList<>();
+        updateSearchAlbumUI(list, album);
+    }
+
+    private void updateSearchAlbumUI(List<Album> list, List<Album> album) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchOfflineFragment.java:159]: ERROR: [PMD-CPD] 64 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/searchonline/SearchOnlineFragment.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchOfflineFragment.java [hunk 4]
```diff
@@ -25,13 +44,68 @@ public void onCreate(Bundle savedInstanceState) {
...
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
             @Nullable Bundle savedInstanceState) {
-        return inflater.inflate(R.layout.fragment_searchoffline, container, false);
+        View view = inflater.inflate(R.layout.fragment_searchoffline, container, false);
+        initViews(view);
+        handleEvents();
+        return view;
+    }
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchOfflineFragment.java:59]: ERROR: [PMD-CPD] 29 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/searchonline/SearchOnlineFragment.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchAlbumRecyclerAdapter.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        extends RecyclerView.Adapter<SearchAlbumRecyclerAdapter.RecyclerViewHolder> {
+
+    private List<Album> mAlbums = new ArrayList<>();
+    private Context mContext;
+
+    public SearchAlbumRecyclerAdapter(Context context) {
+        mContext = context;
+    }
+
+    public void updateData(List<Album> albums) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchAlbumRecyclerAdapter.java:30]: ERROR: [PMD-CPD] 37 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/searchonline/SearchAlbumRecyclerAdapter.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchArtistRecyclerAdapter.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+import com.fstyle.androidtrainning.R;
+import com.fstyle.androidtrainning.model.Artist;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Administrator on 12/15/17.
+ */
+
+public class SearchArtistRecyclerAdapter
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchArtistRecyclerAdapter.java:18]: ERROR: [PMD-CPD] 30 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/searchonline/SearchArtistRecyclerAdapter.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchSongRecyclerAdapter.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    public SearchSongRecyclerAdapter(Context context) {
+        mContext = context;
+    }
+
+    public void updateData(List<Track> tracks) {
+        if (tracks == null) {
+            return;
+        }
+        mTracks.clear();
+        mTracks = tracks;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchSongRecyclerAdapter.java:33]: ERROR: [PMD-CPD] 31 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/searchonline/SearchSongRecyclerAdapter.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchSongRecyclerAdapter.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+            mTxtNameSinger = itemView.findViewById(R.id.text_lower);
+        }
+
+        public void bind(int position) {
+            setNameArtist(position);
+            setNameSong(position);
+        }
+
+        private void setNameArtist(int position) {
+            String nameArtist = mTracks.get(position).getNameArtist();
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchSongRecyclerAdapter.java:80]: ERROR: [PMD-CPD] 20 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/searchonline/SearchSongRecyclerAdapter.java

**Comment by tranminhtri911** [comment]
Ok

**Pull request merged**